docs/src/main/sphinx/language/types.rst

```diff
@@ -5,10 +5,31 @@ Data types
 Trino has a set of built-in data types, described below.
 Additional types can be provided by plugins.
 
-.. note::
+.. _type-mapping-overview:
+
+Trino type support and mapping
+------------------------------
+
+Connectors to data sources are not required to support all Trino data types
+described on this page. If there are data types similar to Trino's that are used
+on the data source, the connector may map the Trino and remote data types to
+each other as needed.
+
+Depending on the connector and the data source, type mapping may apply
+in either direction as follows:
+
+* **Data source to Trino** mapping applies to :ref:`read <sql-read-operations>`
+  and :ref:`write operations <sql-write-operations>`, when
+  columns with specific types in the data source exist and are read or written
+  from Trino. Multiple remote data types may map to the same Trino data type.
+
+* **Trino to data source** mapping applies to any SQL statements that create
+  columns on the data source (such as :doc:`/sql/create-table` and
+  :doc:`/sql/alter-table`). Multiple Trino data types may map to the same remote
+  data type.
```

Comment:
Not exactly correct. It doesn't depend at all on whether columns already exist or not.

`source -> trino` means that any value which gets read from source and needs to be represented in Trino (most commonly SELECT queries).

`trino -> source` means any value in Trino which needs to be represented in remote source (CTAS, INSERT, WHERE predicates etc.)

Reply:
Yeah ... we need to update this @jhlodin  .. turns out I was misunderstanding how this works .. and I am still not sure I get it for the details ..

Reply:
Interesting. How about, as a rewrite:

> **Data source to Trino** mapping applies to any operation where columns in the data source are read by Trino, such as > a `SELECT` statement, and the underlying source data type needs to be represented by a Trino data type.

> **Trino to data source** mapping applies to write operations and predicates where the SQL statement written in Trino
> must be translated into typing that is compatible with the underlying data source. For example, `CREATE TABLE AS` and > `INSERT` statements specify Trino types that are then mapped to remote types on the data source. Predicates like
> `WHERE` also use Trino to data source mappings in order to ensure that the predicate's syntax can resolve on the
> remote data source.

Does this make any more sense?

Reply:
Much better. But I'd still recommend to avoid saying "write operations" for the "Trino to data source".

Probably something like `applies to any operation where the columns or expressions in Trino need to be translated into data types or expressions compatible with the underlying data source.`

Reply:
Applied edits & your suggestion, PTAL